oak_runtime/src/node/http/mod.rs

```diff
@@ -465,56 +528,139 @@ fn get_oak_label(req: &HttpRequest) -> Result<Label, OakStatus> {
     match headers {
         (Some([json_label]), None) => parse_json_label(json_label),
         (None, Some([protobuf_label])) => parse_protobuf_label(protobuf_label),
-        _ => {
-            warn!(
-                "Exactly one header must be provided as an {} or {} header.",
-                oak_abi::OAK_LABEL_HTTP_JSON_KEY,
-                oak_abi::OAK_LABEL_HTTP_PROTOBUF_KEY
-            );
-            Err(OakStatus::ErrInvalidArgs)
-        }
+        _ => Err(HttpError::RequestLabelError(format!(
+            "Exactly one header must be provided as an {} or {} header.",
+            oak_abi::OAK_LABEL_HTTP_JSON_KEY,
+            oak_abi::OAK_LABEL_HTTP_PROTOBUF_KEY
+        ))),
+    }
+}
+
+/// Similar to the request label headers, signed challenge headers can either be JSON formatted or
+/// protobuf encoded. At most one of these formats should be provided. This method:
+/// (1) parses the signed challenge, if one is provided
+/// (2) verifies that the signature is valid
+/// (3) if the signature is valid, returns the public key in the signed challenge as the user's
+/// identity.
+///
+/// User identification is currently an optional feature, so if a challenge response is
+/// not provided, an empty vector is returned.
+fn get_user_identity(req: &HttpRequest) -> Result<Vec<u8>, HttpError> {
+    let headers = (
+        req.headers.as_ref().and_then(|map| {
+            map.headers
+                .get(oak_abi::OAK_SIGNED_CHALLENGE_JSON_KEY)
+                .map(|m| m.values.as_slice())
+        }),
+        req.headers.as_ref().and_then(|map| {
+            map.headers
+                .get(oak_abi::OAK_SIGNED_CHALLENGE_PROTOBUF_KEY)
+                .map(|m| m.values.as_slice())
+        }),
+    );
+
+    match headers {
+        (Some([json_signature]), None) => verify_json_challenge(json_signature),
+        (None, Some([protobuf_signature])) => verify_protobuf_challenge(protobuf_signature),
+        _ => Ok(vec![]),
```

Comment:
Perhaps an error will be easier to debug if both headers are specified, rather than just returning public untrusted.

Reply:
Yes, I agree that the identity should be optional.

Reply:
Thanks. Updated the doc comment. Also, changed the function to return an error when multiple signatures are provided via a single header (the headers can take multiple values).